packages/backend/src/core/totalSupply/getTotalSupplyConfigHash.ts

```diff
@@ -0,0 +1,31 @@
+import { hashJson } from '@l2beat/shared-pure'
+import { sortBy } from 'lodash'
+
+import { TotalSupplyTokensConfig } from './TotalSupplyTokensConfig'
+
+// Increment this value to change the hash which in turn causes the system to
+// recalculate total supplies
+const TOTAL_SUPPLY_LOGIC_VERSION = 0
+
+export function getTotalSupplyConfigHash(
+  totalSupplyTokensConfig: TotalSupplyTokensConfig[],
+) {
+  return hashJson([
+    getEntries(totalSupplyTokensConfig),
+    TOTAL_SUPPLY_LOGIC_VERSION,
+  ])
+}
+
+export function getEntries(totalSupplyTokensConfig: TotalSupplyTokensConfig[]) {
+  const serializedEntries = totalSupplyTokensConfig.map(
+    ({ assetId, tokenAddress, sinceTimestamp, decimals }) => {
+      return {
+        assetId: assetId.toString(),
+        tokenAddress: tokenAddress.toString(),
+        sinceTimestamp: sinceTimestamp.toNumber(),
+        decimals: decimals,
+      }
+    },
+  )
+  return sortBy(serializedEntries, ['assetId', 'tokenAddress', 'decimals'])
+}
```

Comment:
Ported logic validation also applies here, PTL

Reply:
also some tests would come in handy

Reply:
probably we would also like some notion of project (projectId) in this hash

Reply:
If I understand correctly, it should not be a problem. We have a per-chain instance of total supply updater and `chainId` serves as a required discriminator to prevent collisions/overlaps.

Reply:
https://github.com/l2beat/l2beat/pull/1664/commits/f44b6d2d499a7a07c74a20e38f053f8160066655